test/field_exchanger_tests.jl

```diff
@@ -0,0 +1,192 @@
+using Test
+using ClimaCore: Meshes, Domains, Topologies, Spaces, Fields, InputOutput
+using ClimaCoupler: Utilities, Regridder, TestHelper
+import ClimaCoupler.Interfacer:
+    update_field!, AtmosModelSimulation, SurfaceModelSimulation, SurfaceStub, get_field, update_field!
+import ClimaCoupler.FluxCalculator: CombinedAtmosGrid, PartitionedComponentModelGrid
+import ClimaCoupler.FieldExchanger:
+    import_atmos_fields!,
+    import_combined_surface_fields!,
+    update_model_sims!,
+    reinit_model_sims!,
+    reinit!,
+    step_model_sims!,
+    step!
+
+FT = Float64
+
+# test for a simple generic atmos model
+struct DummySimulation{C} <: AtmosModelSimulation
+    cache::C
+end
+
+get_field(sim::DummySimulation, ::Val{:turbulent_energy_flux}) = sim.cache.turbulent_energy_flux
+get_field(sim::DummySimulation, ::Val{:turbulent_moisture_flux}) = sim.cache.turbulent_moisture_flux
+get_field(sim::DummySimulation, ::Val{:radiative_energy_flux}) = sim.cache.radiative_energy_flux
+get_field(sim::DummySimulation, ::Val{:liquid_precipitation}) = sim.cache.liquid_precipitation
+get_field(sim::DummySimulation, ::Val{:snow_precipitation}) = sim.cache.snow_precipitation
+
+@testset "import_atmos_fields!" begin
+
+    boundary_space = TestHelper.create_space(FT)
+    coupler_names = (:F_A, :F_E, :F_R, :P_liq, :P_snow)
+    atmos_names = (
+        :turbulent_energy_flux,
+        :turbulent_moisture_flux,
+        :radiative_energy_flux,
+        :liquid_precipitation,
+        :snow_precipitation,
+    )
+    atmos_fields = NamedTuple{atmos_names}(ntuple(i -> Fields.ones(boundary_space), length(coupler_names)))
+
+    model_sims = (; atmos_sim = DummySimulation(atmos_fields))
+
+    flux_types = (CombinedAtmosGrid(), PartitionedComponentModelGrid())
```

Comment:
Could you add a comment explaining what behavior we expect to see with each of the flux types? :)

Reply:
Sure, I clarified this in the docs, and added comments to the tests.